csrc/device_lower/pass/index.cpp

```diff
@@ -1887,37 +1887,27 @@ void IndexLowering::handle(const PadOp* pad) {
 
   const auto pad_val = pad->value();
 
-  std::unordered_map<IterDomain*, Val*> override_index;
```

Comment:
This was necessary because a broadcast domain gets 0 as its index, no matter what the index of the corresponding consumer domain is.

Reply:
Ah yes, this was needed because of using producer index in the pad predicate. Now that we are using consumer index, it's not needed. That's great. I believe `getIterationIndexForBroadcast` above can now be removed.